nix/default.nix

```diff
@@ -54,6 +54,7 @@ let
   # for cabal, as setting it in direnv can interfere with programs in the host
   # system, especially for non-NixOS users.
   cabal-wrapper = pkgs.writeShellScriptBin "cabal" ''
+    export NIX_BUILD_SHELL="${pkgs.bash}/bin/bash"
```

Comment:
separate PR?  it's ok if you're lazy and somebody approves, just couldn't keep myself from raising an eyebrow at least.  :)